clusters/resource_cluster.go

```diff
@@ -156,93 +206,105 @@ func resourceClusterSchema() map[string]*schema.Schema {
 }
 
 func resourceClusterCreate(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-	var cluster Cluster
 	start := time.Now()
 	timeout := d.Timeout(schema.TimeoutCreate)
-	clusters := NewClustersAPI(ctx, c)
+	w, err := c.WorkspaceClient()
+	if err != nil {
+		return err
+	}
+	clusters := w.Clusters
+	var cluster compute.CreateCluster
 	common.DataToStructPointer(d, clusterSchema, &cluster)
-	if err := cluster.Validate(); err != nil {
+	if err := Validate(cluster); err != nil {
+		return err
+	}
+	ModifyRequestOnInstancePool(&cluster)
+	if cluster.Autoscale == nil {
+		cluster.ForceSendFields = []string{"NumWorkers"}
+	}
+	clusterWaiter, err := clusters.Create(ctx, cluster)
+	if err != nil {
 		return err
 	}
-	cluster.ModifyRequestOnInstancePool()
-	// TODO: propagate d.Timeout(schema.TimeoutCreate)
-	clusterInfo, err := clusters.Create(cluster)
+	clusterInfo, err := clusterWaiter.GetWithTimeout(timeout)
 	if err != nil {
 		return err
 	}
-	d.SetId(clusterInfo.ClusterID)
-	d.Set("cluster_id", clusterInfo.ClusterID)
+	d.SetId(clusterInfo.ClusterId)
+	d.Set("cluster_id", d.Id())
 	isPinned, ok := d.GetOk("is_pinned")
 	if ok && isPinned.(bool) {
-		err = clusters.Pin(clusterInfo.ClusterID)
+		err = clusters.PinByClusterId(ctx, d.Id())
 		if err != nil {
 			return err
 		}
 	}
-	var libraryList libraries.ClusterLibraryList
+
+	var libraryList compute.InstallLibraries
 	common.DataToStructPointer(d, clusterSchema, &libraryList)
```

Comment:
The point of adding libraries to the schema was that we don't need to call `DataToStructPointer` multiple times, right?
```suggestion
	libraryList := compute.InstallLibraries{
	    ClusterId: clusterInfo.ClusterId,
	    Libraries: cluster.Libraries,
	}
```